### PR TITLE
refs #29271 - template macros return array instead of string

### PR DESCRIPTION
**Pull request by domitea**
Fixing correct result for several macros. Current implementation takes values from array and join them together. This fix remain array of values in array.

### CODE REVIEW

File: app/lib/katello/concerns/base_template_scope_extensions.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def host_products(host)
       def host_products_names(host)
         products = host_products(host)
         if products
-          products.map(&:name).join(', ')
+          products.map(&:name)
         else
           ''
```
Comment (tbrisker) [inline app/lib/katello/concerns/base_template_scope_extensions.rb]: should this be `[]` instead?

Comment (jturel) [reply app/lib/katello/concerns/base_template_scope_extensions.rb]: Likely the else can be removed entirely

**Comment by jturel** [comment]
@domitea is the intent here that the templates will be responsible for turning the arrays into comma-separated strings?

File: app/lib/katello/concerns/base_template_scope_extensions.rb (hunk 1 above)
Comment (tbrisker) [reply app/lib/katello/concerns/base_template_scope_extensions.rb]: i think the condition is needed because `host_products` can return nil

**Comment by domitea** [comment]
@jturel yes, it is.

File: app/lib/katello/concerns/base_template_scope_extensions.rb (hunk 1 above)
Comment (jturel) [reply app/lib/katello/concerns/base_template_scope_extensions.rb]: Oh :confounded:

**Comment by tbrisker** [comment]
test failures are unrelated, due to added rubocop plugin (https://community.theforeman.org/t/katello-master-source-release-850-failed/17902)

**Comment by jturel** [comment]
[test katello]

**Comment by jturel** [comment]
[test katello]

**Pull request merged**